mitmproxy/http.py

```diff
@@ -469,6 +469,8 @@ def decode(self, strict: bool = True) -> None:
         *Raises:*
          - `ValueError`, when the content-encoding is invalid and strict is True.
         """
+        if not self.raw_content:
+            return
```

Comment:
Let's add tests for this in test_http.py:

1. One response with missing body (e.g. body streaming)
2. One response with empty body (e.g. a HEAD response)